crates/full-node/sov-ethereum/src/handlers/mod.rs

```diff
@@ -208,6 +211,13 @@ pub(crate) mod signer {
             )?;
             transaction_request.gas = Some(estimated_gas.to::<u64>());
 
+            // For contract deployments, convert `to: None` to `to: Some(TxKind::Create)`
+            // The JSON-RPC spec uses `null` or omitted `to` field for contract deployments,
+            // but alloy's `build_typed_tx()` requires `Some(TxKind::Create)`
+            if transaction_request.to.is_none() {
```

Comment:
Don't we need the same for eth_sendRawTransaction?

Reply:
sendRawTransaction does not construct the TX from TxRequest but gets an already constructed TX and therefore does not trigger this code path in alloy